src/test/unit/StrategyManagerUnit.t.sol

```diff
@@ -532,187 +897,241 @@ contract StrategyManagerUnitTests is Test, Utils {
+        uint256 removeSharesAmount
+    ) external {
+        cheats.assume(staker != address(0));
+        cheats.assume(depositAmount > 0 && depositAmount < dummyToken.totalSupply());
+        cheats.assume(removeSharesAmount > depositAmount);
         IStrategy strategy = dummyStrat;
-        IERC20 token = dummyToken;
-        uint256 amount = 1e18;
-
-        reenterer.prepareReturnData(abi.encode(uint256(0)));
-
-        cheats.startPrank(staker);
-        cheats.expectRevert(bytes("StrategyManager._addShares: shares should not be zero!"));
-        strategyManager.depositIntoStrategy(strategy, token, amount);
-        cheats.stopPrank();
+        _depositIntoStrategySuccessfully(strategy, staker, depositAmount);
+        cheats.expectRevert(bytes("StrategyManager._removeShares: shareAmount too high"));
+        delegationManagerMock.removeShares(strategyManager, staker, strategy, removeSharesAmount);
     }
 
-    function test_addSharesRevertsWhenDepositWouldExeedMaxArrayLength() external {
-        address staker = address(this);
-        IERC20 token = dummyToken;
-        uint256 amount = 1e18;
+    /**
+     * @notice testing removeShares()
+     * deposits 1 strategy and tests it is removed from staker strategy list after removing all shares
+     */
+    function testFuzz_RemovesStakerStrategyListSingleStrat(address staker, uint256 sharesAmount) external {
+        cheats.assume(staker != address(0));
+        cheats.assume(sharesAmount > 0 && sharesAmount < dummyToken.totalSupply());
         IStrategy strategy = dummyStrat;
+        _depositIntoStrategySuccessfully(strategy, staker, sharesAmount);
 
-        // uint256 MAX_STAKER_STRATEGY_LIST_LENGTH = strategyManager.MAX_STAKER_STRATEGY_LIST_LENGTH();
-        uint256 MAX_STAKER_STRATEGY_LIST_LENGTH = 32;
-
-        // loop that deploys a new strategy and deposits into it
-        for (uint256 i = 0; i < MAX_STAKER_STRATEGY_LIST_LENGTH; ++i) {
-            cheats.startPrank(staker);
-            strategyManager.depositIntoStrategy(strategy, token, amount);
-            cheats.stopPrank();
-
-            dummyStrat = deployNewStrategy(dummyToken, strategyManager, pauserRegistry, dummyAdmin);
-            strategy = dummyStrat;
-
-            // whitelist the strategy for deposit
-            cheats.startPrank(strategyManager.owner());
-            IStrategy[] memory _strategy = new IStrategy[](1);
-            _strategy[0] = dummyStrat;
-            cheats.expectEmit(true, true, true, true, address(strategyManager));
-            emit StrategyAddedToDepositWhitelist(dummyStrat);
-            strategyManager.addStrategiesToDepositWhitelist(_strategy);
-            cheats.stopPrank();
-        }
+        uint256 stakerStrategyListLengthBefore = strategyManager.stakerStrategyListLength(staker);
+        uint256 sharesBefore = strategyManager.stakerStrategyShares(staker, strategy);
+        require(sharesBefore == sharesAmount, "Staker has not deposited amount into strategy");
```

Comment:
assertEq() instead of require.  The trace is nicer and outputs similar to below:
```
left : shows value
right : shows value
optional reason string
```

Reply:
in the case the equality is violated*

Reply:
Changed requires into asserts https://github.com/Layr-Labs/eigenlayer-contracts/pull/293